### ISSUE TITLE
Unformatted format string in error message output

### ISSUE THREAD

**Issue by nh2** [NixOS/nix#3578]
**Describe the bug**

Today I got this output (with nix `2.3.2`):

```
nar upload failed: sha256-1iFK6O3yD4qHGeSEl4RDOhh3+Hsk6PFzh+DKrXK1JSY= Internal Server Error
error: failed to add path '%s' to remote host '%s', info.path, host
```

That looks like an unformatted format string, it looks like it's somehow printing the code instead of running the code.

That seems to be this line https://github.com/NixOS/nix/blame/ebd89999c28e25e71048d523f9c4f8b856dee9a9/src/libstore/legacy-ssh-store.cc#L178 but I don't quite understand what's going on, it doesn't look wrong. Are some macros involved?

It didn't reoccur with the next build so I cannot currently reproduce it.

**Comment by edolstra** [comment]
This was fixed in bbe97dff8b3054d96e758f486f9ce3fa09e64de3.

**Comment by nh2** [comment]
Thanks!
